test/test_nn_ops.cpp

```diff
@@ -106,46 +106,198 @@ TEST(NNOps, BiasAddGrad) {
   opexecuter_4D.RunTest();
 }
 
-TEST(NNOps, Conv2DBackpropFilterNHWC) {
+// TF does not support NCHW kernels
+// To test NCHW data format
+// Create graph with inputs in NCHW format and execute on NGraph
+// Reshape the NCHW inputs to NHWC and run on TF
+// Compare the results
+TEST(tf_exec, Conv2DBackpropFilterNCHWSame) {
+  string padding_type = "SAME";
+  // Filter :[filter_height, filter_width, in_channels, out_channels]
+  initializer_list<int> filter_size_HWIO = {3, 3, 2, 2};
+  vector<DataType> output_datatypes = {DT_FLOAT};
+  vector<int> static_input_indexes = {1};
+
+  // Define scope for nGraph
+  // Data Format : NCHW
+  Scope ngraph_scope = Scope::NewRootScope();
+  vector<int64> input_size_NCHW = {1, 2, 7, 6};
+  Tensor input_data_NCHW(DT_FLOAT, TensorShape(input_size_NCHW));
+  AssignInputValuesAnchor(input_data_NCHW, -1.1f);
+
+  vector<int64> output_del_size_NCHW = {1, 2, 4, 3};
+  Tensor output_delta_NCHW(DT_FLOAT, TensorShape(output_del_size_NCHW));
+  AssignInputValuesAnchor(output_delta_NCHW, -1.1f);
+
+  auto filter_sizes = ops::Const(ngraph_scope, filter_size_HWIO);
+  vector<int> stride_NCHW = {1, 1, 2, 2};
+
+  // Dialtion rates > 1 not supported by TF
+  ops::Conv2DBackpropFilter::Attrs op_attr_nchw;
+  op_attr_nchw = op_attr_nchw.DataFormat("NCHW");
+  op_attr_nchw = op_attr_nchw.Dilations({1, 1, 1, 1});
+
+  auto r_ngraph = ops::Conv2DBackpropFilter(
+      ngraph_scope, input_data_NCHW, filter_sizes, output_delta_NCHW,
+      stride_NCHW, padding_type, op_attr_nchw);
+
+  vector<Output> sess_run_fetchoutputs = {r_ngraph};
+  OpExecuter opexecuter_ngraph(ngraph_scope, "Conv2DBackpropFilter",
+                               static_input_indexes, output_datatypes,
+                               sess_run_fetchoutputs);
+  vector<Tensor> ngraph_outputs;
+  opexecuter_ngraph.ExecuteOnNGraph(ngraph_outputs);
+
+  // Define scope for tf (without nGraph)
+  // Data Format: NHWC
+  Scope tf_scope = Scope::NewRootScope();
+  auto input_data_NHWC =
+      ops::Transpose(tf_scope, input_data_NCHW, {0, 2, 3, 1});
+  auto output_delta_NHWC =
+      ops::Transpose(tf_scope, output_delta_NCHW, {0, 2, 3, 1});
+  auto filter_sizes_tf = ops::Const(tf_scope, filter_size_HWIO);
+  vector<int> stride_NHWC = {1, 2, 2, 1};
+  auto r_tf =
+      ops::Conv2DBackpropFilter(tf_scope, input_data_NHWC, filter_sizes_tf,
+                                output_delta_NHWC, stride_NHWC, padding_type);
+  vector<Output> sess_run_fetchoutputs_tf = {r_tf};
+  OpExecuter opexecuter_tf(tf_scope, "Conv2DBackpropFilter",
+                           static_input_indexes, output_datatypes,
+                           sess_run_fetchoutputs_tf);
+
+  vector<Tensor> tf_outputs;
+  opexecuter_tf.ExecuteOnTF(tf_outputs);
+
+  // Compare NGraph and TF Outputs
+  OpExecuter::CompareNGraphAndTF(tf_outputs, ngraph_outputs);
+}
```

Comment:
so we have:
```
stuff
vector<Tensor> ngraph_outputs;
opexecuter_ngraph.ExecuteOnNGraph(ngraph_outputs);
stuff
vector<Tensor> tf_outputs;
opexecuter_tf.ExecuteOnTF(tf_outputs);
OpExecuter::CompareNGraphAndTF(tf_outputs, ngraph_outputs);
```

but, ngraph_outputs is just ngraph_outputs_
and tf_outputs is just tf_outputs_
so we pull them out of the class and then send them into CompareNGraphAndTF. That sounds roundabout.

So we are creating 2 opexecutors. instead how about:
```
create opexp
stuff
opexp.ExecuteOnNGraph();
opexp.clear_graph()
opexp.set_new_graph() . // a setter for graph or whatever needs to change
stuff
opexp.ExecuteOnTF();
OpExecuter::CompareNGraphAndTF();
```

Even if we do not do the above  rearraggement (setter for new graph/scope), I think the (getter for tf_outputs_ and ngraph_outputs_ + no overloading) would help.